### ISSUE TITLE
Cropped output on large monitor

### ISSUE THREAD

**Issue by sbihel** [alacritty/alacritty#3695]
### Issue
On a large 4k monitor, if I open a program that takes the whole screen like Vim, Mutt, or Tmux, the bottom of the screen stays black (my background colour).

It doesn't seem to be related to #824 as key presses will not do a complete redraw and the number of lines displayed is consistent.

The interesting thing is, if I reduce the window vertically (i.e. reduce the number of lines) the same number of lines will be displayed, but if I reduce the window horizontally, more and more lines are displayed. If I increase the font size, more lines are displayed.

I did a test by filling the screen with `y` and it seems that the maximum number of `y`s displayed is 65536 (2^16).

So it makes me think that it went past the maximum number of characters (cells? glyphs?) to be displayed. I tried to look in the code to see if there was a type that wasn't big enough but to no avail. If you have any hint, I'm happy to invest more time.

Or maybe it is intended to be like this? It's too bad because I like to have Tmux in fullscreen, maybe a warning in the README or in the program would be nice.

Any way, thanks a lot for Alacritty! ❤️ 

### System

OS: macOS
Version: `alacritty 0.4.2 (f68de37)` and `alacritty 0.5.0-dev (81ce935)`

### Logs

Font/Terminal size:
```
Device pixel ratio: 1
Cell Size: 5 x 11
Padding: 0 x 0
New num_cols is 768 and num_lines is 196
Width: 3840, Height: 2160
Font size changed to Size(18) with DPR of 1
```

**Comment by kchibisov** [comment]
>I did a test by filling the screen with y and it seems that the maximum number of ys displayed is 65536 (2^16).

I can use 3840x2160 grid on my monitor just fine (1 pixel per terminal cell) just fine. Which is way larger than what you're testing here. It'll be really strange that we have u16 limit somewhere, which can only be reached by you.

**Comment by chrisduerr** [comment]
Could you share a screenshot of your problem? It almost just sounds like you're describing the differences in padding at the top/bottom, which is normal.

**Comment by sbihel** [comment]
![Screenshot 2020-05-08 at 09 34 16](https://user-images.githubusercontent.com/12815410/81387775-2b95dc80-910f-11ea-8cd7-aaea55a8c441.png)

For the `window` settings, I only set `decorations: none` but the issue also occurs without a config file.

**Comment by chrisduerr** [comment]
@sbihel Is Alacritty frozen? Can you select the text or the empty space below the text?

**Comment by sbihel** [comment]
It's not frozen no. I can select text, move the window, type, etc. Everything is working... it's just not showing the bottom.

**Comment by kchibisov** [comment]
I feel like macOS just breaks at some point and no-ops on OpenGL calls, idk.

**Comment by chrisduerr** [comment]
Maybe a memory/VRAM issue? Though I cannot imagine that you have less than the ~20MB or something required to run Alacritty.

What happens when you have multiple terminals of that size open @sbihel? Does it get worse?

**Comment by sbihel** [comment]
Thanks for the help!

Having multiple windows/terminals doesn't have any effect. And I have 16GB of memory, which seems to be healthy in my experience.

I did a `git bisect` and it seems that 40237b213a847cb3fabaa4da0ef2f295e9bf56a9 introduced the bug. I had a look but couldn't identify the problematic change. I'll try to spend more time if no one beats me to it.

**Comment by chrisduerr** [comment]
Could you share additional system information like GPU and macOS version?

**Comment by sbihel** [comment]
MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports)
OS: macOS Catalina 10.15.4
Processor: 2.5 GHz Dual-Core Intel Core i7
Memory: 16 GB 2133 MHz LPDDR3
Graphics: Intel Iris Plus Graphics 640 1536 MB

**Comment by chrisduerr** [comment]
I'll have to test this out on my macbook to see if it might be macOS specific.

Maybe @casperstorm also wants to give it a try?

**Comment by chrisduerr** [comment]
I can reproduce and this really looks like a driver bug.

**Comment by sbihel** [comment]
Apple deprecated the use of OpenGL in favour of Metal. Do you think it's related?

**Comment by chrisduerr** [comment]
No, there was no active removal of OpenGL functionality. They just marked the API as deprecated. Apple's support for OpenGL was never great though.

**Comment by dminca** [comment]
Hi, just dropping by to say I'm currently facing the same on MacOS Sonoma 14.5

My spec

```
MacOS Sonoma 14.5 (23F79)
Processor: 2,3 GHz Quad-Core Intel Core i7
Graphics: Intel Iris Plus Graphics 1536 MB
Memory: 32 GB 3733 MHz LPDDR4X
```

with Alacritty version `0.13.2`
